### PR TITLE
ref: add method to SentryThreadWrapper to ensure code runs on main thread

### PR DESCRIPTION
**Pull request by armcknight**
And use to schedule the NSTimer in SentryTracer on the main thread.

#skip-changelog

### CODE REVIEW

File: Sources/Sentry/SentryThreadWrapper.m [hunk 1]
```diff
@@ -19,6 +19,15 @@ - (void)threadFinished:(NSUUID *)threadID
     // No op. Only needed for testing.
 }
 
++ (void)onMainThread:(void (^)(void))block
```
Comment (philipphofmann) [inline Sources/Sentry/SentryThreadWrapper.m:22]: `m`: A test for this would be nice.

**Review by philipphofmann — APPROVED** [review]
LGTM

File: Sources/Sentry/SentryThreadWrapper.m (hunk 1 above)
Comment (armcknight) [reply Sources/Sentry/SentryThreadWrapper.m:22]: done!

File: Tests/SentryTests/Helper/SentryThreadWrapperTests.swift [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import XCTest
+
+final class SentryThreadWrapperTests: XCTestCase {
```
Comment (philipphofmann) [inline Tests/SentryTests/Helper/SentryThreadWrapperTests.swift:3]: 💯

File: Tests/SentryTests/Helper/SentryThreadWrapperTests.swift [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    func testOnMainThreadFromMainThread() {
+        let e = expectation(description: "Asserted that execution happened on main thread")
+        SentryThreadWrapper.onMainThread {
+            XCTAssertTrue(Thread.isMainThread)
+            e.fulfill()
+        }
+        waitForExpectations(timeout: 1)
+    }
+
+    func testOnMainThreadFromNonmainContext() {
```
Comment (philipphofmann) [inline Tests/SentryTests/Helper/SentryThreadWrapperTests.swift:13]: `l`: Shouldn't it be `NonMain`?
```suggestion
    func testOnMainThreadFromNonMainContext() {
```

**Review by philipphofmann — APPROVED** [review]
LGTM

**Pull request merged**